chia/full_node/block_store.py

```diff
@@ -358,17 +373,29 @@ async def get_block_records_in_range(
         if present.
         """
 
-        formatted_str = f"SELECT header_hash, block from block_records WHERE height >= {start} and height <= {stop}"
-
-        cursor = await self.db.execute(formatted_str)
-        rows = await cursor.fetchall()
-        await cursor.close()
         ret: Dict[bytes32, BlockRecord] = {}
-        for row in rows:
-            # TODO: address hint error and remove ignore
-            #       error: Invalid index type "bytes" for "Dict[bytes32, BlockRecord]"; expected type "bytes32"  [index]
-            header_hash = self.maybe_from_hex(row[0])
-            ret[header_hash] = BlockRecord.from_bytes(row[1])  # type: ignore[index]
+        if self.db_wrapper.db_version == 2:
+
+            cursor = await self.db.execute(
+                "SELECT header_hash, block_record FROM full_blocks WHERE height >= ? AND height <= ?",
+                (start, stop),
+            )
+            rows = await cursor.fetchall()
+            await cursor.close()
+            for row in rows:
+                header_hash = bytes32(row[0])
+                ret[header_hash] = BlockRecord.from_bytes(row[1])
```

Comment:
Maybe?  This should distribute the async bit and avoid larger blocks of sync activity.  It also avoids forcing the whole result set into memory at one time in `rows`.  Though sure, it does end up in memory in `ret`, but at least it isn't twice.

All that said, maybe there's some relevant detrimental performance effect here.  I don't know offhand.

Again, if this change is wanted, it is present in multiple places.

```suggestion
            async with self.db.execute(
                "SELECT header_hash, block_record FROM full_blocks WHERE height >= ? AND height <= ?",
                (start, stop),
            ) as cursor:
                async for row in cursor:
                    header_hash = bytes32(row[0])
                    ret[header_hash] = BlockRecord.from_bytes(row[1])
```

Reply:
I agree with this also, and would prefer changing this in a separate PR. In my mind, the main benefit is to allow pipelining of the work. We can parse `BlockRecord` concurrently with reading data from disk.